packages/app-desktop/gui/dialogs.ts

```diff
@@ -21,14 +21,12 @@ class Dialogs {
 		}
 	}
 
-	public async prompt(message: string, title = '', defaultValue = '', options: any = null) {
-		if (options === null) options = {};
-
+	public async prompt(message: string, title = '', defaultValue = '', options: any = { cancel: true }) {
```

Comment:
Can you provide the smalltalk doc for this? I don't understand what passing "cancel" to the prompt call could mean.

Reply:
Smalltalk doc is not extensive and therefore they do not provide good explanations of what does the cancel option do. In their [README (v2.5.1)](https://github.com/coderaiser/smalltalk/tree/v2.5.1) they state the following:

> In every method of smalltalk last parameter options is optional and could be used for preventing of handling cancel event.
> 
> ```java
> {
>     cancel: true /* default */
> }
> ```

But this statement is useless because:
1. true is not the default option;
2. it is not clear what they mean when they say "preventing of handling cancel event".

So to better understand you need to [dig deeper and read their source code](https://github.com/coderaiser/smalltalk/blob/v2.5.1/lib/smalltalk.js), where you can find the promise that is returned when creating any dialog:
```java
const promise = new Promise((resolve, reject) => {
    const noCancel = options && !options.cancel;
    const empty = () => {};

    ok(resolve);
    cancel(noCancel ? empty : reject);
});
```

There you can see that reject() is only called if noCancel is false, otherwise they call an empty function.

Reply:
Ok thank you for clarifying. Would you mind doing this?

- Leave `options: any = null` by default
- Add below `options = { cancel: true, ...options }` (this is the pattern we prefer for options)
- Above the smalltalk call, please add a link to your comment here for more explanation